src/test/rgw/sfs/test_rgw_sfs_sfs_bucket.cc

```diff
@@ -1453,3 +1497,191 @@ TEST_F(TestSFSBucket, ListNamespaceMultipartsBasics) {
+  rgw::sal::Attrs arg_attrs;
+
+  RGWBucketInfo arg_info = get_binfo();
+  obj_version arg_objv;
+  bool existed = false;
+  req_info arg_req_info(ceph_context.get(), &env);
+
+  std::unique_ptr<rgw::sal::Bucket> bucket_from_create;
+
+  EXPECT_EQ(
+      user->create_bucket(
+          &ndp,                    //dpp
+          arg_bucket,              //b
+          "zg1",                   //zonegroup_id
+          arg_pl_rule,             //placement_rule
+          arg_swift_ver_location,  //swift_ver_location
+          &arg_quota_info,         //pquota_info
+          arg_aclp_d,              //policy
+          arg_attrs,               //attrs
+          arg_info,                //info
+          arg_objv,                //ep_objv
+          false,                   //exclusive
+          false,                   //obj_lock_enabled
+          &existed,                //existed
+          arg_req_info,            //req_info
+          &bucket_from_create,     //bucket
+          null_yield               //optional_yield
+      ),
+      0
+  );
+
+  std::unique_ptr<rgw::sal::Bucket> bucket_from_store_1;
+
+  EXPECT_EQ(
+      store->get_bucket(
+          &ndp, user.get(), arg_info.bucket, &bucket_from_store_1, null_yield
+      ),
+      0
+  );
+
+  std::unique_ptr<rgw::sal::Bucket> bucket_from_store_2;
+
+  EXPECT_EQ(
+      store->get_bucket(
+          &ndp, user.get(), arg_info.bucket, &bucket_from_store_2, null_yield
+      ),
+      0
+  );
+
+  EXPECT_EQ(*bucket_from_store_1, *bucket_from_store_2);
+
+  // merge_and_store_attrs
+
+  rgw::sal::Attrs new_attrs;
+  RGWAccessControlPolicy arg_aclp = get_aclp_1();
+  {
+    bufferlist acl_bl;
+    arg_aclp.encode(acl_bl);
+    new_attrs[RGW_ATTR_ACL] = acl_bl;
+  }
+
+  EXPECT_EQ(
+      bucket_from_store_1->merge_and_store_attrs(&ndp, new_attrs, null_yield), 0
+  );
+
+  // assert b1 contains the RGW_ATTR_ACL attribute
+  auto acl_bl_1 = bucket_from_store_1->get_attrs().find(RGW_ATTR_ACL);
+  EXPECT_NE(bucket_from_store_1->get_attrs().end(), acl_bl_1);
+
+  // assert b2 does not contain the RGW_ATTR_ACL attribute
+  auto acl_bl_2 = bucket_from_store_2->get_attrs().find(RGW_ATTR_ACL);
+  EXPECT_EQ(bucket_from_store_2->get_attrs().end(), acl_bl_2);
+
+  // put_info
+
+  RGWObjectLock obj_lock;
+  mockable::RGWObjectLock& ol = mockable::actual2mock(obj_lock);
+  ol.enabled = true;
+  ol.rule.defaultRetention.years = 12;
+  ol.rule.defaultRetention.days = 31;
+  ol.rule.defaultRetention.mode = "GOVERNANCE";
+  ol.rule_exist = true;
+
+  bucket_from_store_2->get_info().obj_lock = obj_lock;
+  EXPECT_EQ(bucket_from_store_2->put_info(&ndp, false, real_time()), 0);
+
+  auto& ol1 = mockable::actual2mock(bucket_from_store_1->get_info().obj_lock);
+  auto& ol2 = mockable::actual2mock(bucket_from_store_2->get_info().obj_lock);
+
+  // obj lock structure in memory cannot be equal at this point in memory for the
+  // two b1 and b2 objects; this simulates two threads updating metadata over the
```

Comment:
Why cant they be equal?

Reply:
Per test construction they cannot be equal, `b1` memory is modified with `merge_and_store_attrs() `and `b2` memory is modified with `put_info()`.
Maybe better change in the comment the b1 and b2 aliases with their correct names ( bucket_from_store_1 and bucket_from_store_2) ?

Reply:
I think it is worth adding a bit more of an explanation in this case. What you just replied on this thread, for instance.